src/pages/Admin/Venue/VenueDeleteModal.tsx

```diff
@@ -1,106 +1,64 @@
-    try {
-      await firebase.functions().httpsCallable("venue-deleteVenue")({
-        id: venue.id,
-      });
-      setDeleted(true);
-    } catch (error) {
-      setError(error.message);
-    } finally {
-      setDeleting(false);
-    }
-  }, [venue.id]);
+  const [{ error, loading: isDeleting }, deleteVenue] = useAsyncFn(async () => {
+    await firebase.functions().httpsCallable("venue-deleteVenue")({
+      id: venueId,
+    });
+    onDelete?.();
+  }, [onDelete, venueId]);
 
   return (
-    <Modal show={show} onHide={closeDeleteModal}>
-      <div className="venue-delete-modal-container">
-        <h2 className="centered">Delete venue</h2>
+    <Modal show={show} onHide={onHide}>
+      <div className="VenueDeleteModal">
+        <h2 className="centered">Delete space</h2>
         <div className="secondary-action">
-          WARNING: Permanently removes a venue from SparkleVerse
+          WARNING: This action cannot be undone and will permanently remove the
+          space <b>{venueName}</b>!
         </div>
-        {!deleted && (
-          <>
-            <div className="input-group">
-              <span className="info">
-                WARNING: This action cannot be undone! Are you sure you wish to
-                delete {venue.name}?
-              </span>
+        {isDeleting && (
+          <div className="centered-flex">
+            <div className="spinner-border">
+              <span className="sr-only">Loading...</span>
             </div>
-            {deleting && (
-              <div className="centered-flex" style={{ marginBottom: 10 }}>
-                <div className="spinner-border">
-                  <span className="sr-only">Loading...</span>
-                </div>
-              </div>
-            )}
-            <div className="input-group">
-              <button
-                disabled={deleting}
-                className="btn btn-danger btn-block btn-centered"
-                onClick={deleteVenue}
-              >
-                Yes, Delete
-              </button>
-              {error && <span className="input-error">{error}</span>}
-              <button
-                disabled={deleting}
-                className="btn btn-primary btn-block btn-centered"
-                onClick={closeDeleteModal}
-              >
-                No, Cancel
-              </button>
-            </div>
-          </>
-        )}
-        {deleted && (
-          <div className="input-group">
-            <span className="info">Venue has been permanently deleted.</span>
-            <button
-              className="btn btn-primary btn-block btn-centered"
-              type="submit"
-              onClick={closeDeleteModal}
-            >
-              Close
-            </button>
           </div>
         )}
+        <div className="VenueDeleteModal__buttons">
+          <ButtonNG
+            disabled={isDeleting}
+            variant="danger"
+            onClick={deleteVenue}
+          >
```

Comment:
```suggestion
            loading={isDeleting}
          >
```

Reply:
There is already a spinner above the buttons and it looks weird when I add this flag to the button because there's 2 spinners on the screen. :/

Reply:
Any rule saying we should keep the original spinner? 😄 Probably best left as you've made it.

Which reminds me, the button supports `aria-` props, so it will be prudent to check which might apply as we add/replace.

[OOS]
`sr-only` class should probably be added on ButtonNG for the loading state

Reply:
> Probably best left as you've made it.

I'll leave it as it is for now, but it can be improved, yes.

> Any rule saying we should keep the original spinner?

Not really, we can remove it but there is no design for this...